executor.go

```diff
@@ -42,20 +42,19 @@ func Execute(p ExecuteParams) (result *Result) {
 		if len(extErrs) != 0 {
 			result.Errors = append(result.Errors, extErrs...)
 		}
-
 		addExtensionResults(&p, result)
 	}()
 
 	resultChannel := make(chan *Result, 2)
 
-	go func() {
+	go func(out chan<- *Result, done <-chan struct{}) {
```

Comment:
nit: no need to pass them in explicitly it's easier to see when using `resultChannel` directly.